www/content/workstation/upgrade_labs.md

```diff
@@ -0,0 +1,455 @@
+```
+
+#### Add Cookbook Source Locations
+
+After creating the repo, `chef capture NODE` prompts you to fetch the cookbooks from their original locations.
+
+##### Main Cookbook Development Location
+
+The `chef capture` command prompts you first for your main cookbook source location.
+
+```
+Please clone or check out the following cookbooks locally
+from their original sources, and provide the base path
+for the checkout:
+
+  - cron (v1.6.1)
+  - chef-client (v4.3.0)
+  - logrotate (v1.9.2)
+  - windows (v1.44.1)
+  - chef_handler (v1.4.0)
+
+If all cookbooks are not available in the same base location,
+you will have a chance to provide additional locations.
+
+If sources are not available for these cookbooks, leave this blank.
+
+Checkout Location [none]:
+```
+
+At this point, enter the path to your [cookbook development directory](/#cookbook-directory-layout), for example, `/Users/user_name/my_cookbooks` at the prompt.
+
+
+`chef capture` scans that path and locates the cookbooks that it needs. The command finishes once it accounts for all cookbook sources; but if any are missing, it will prompt for them in the next step.
+
+```
+Checkout Location [none]: /src/my_cookbooks
+  Replacing cookbook: cron
+  Replacing cookbook: chef-client
+```
+
+##### Alternate Cookbook Source Locations
+
+Suppose that your node requires 5 cookbooks:
+
+  - cron (v1.6.1)
+  - chef-client (v4.3.0)
+  - logrotate (v1.9.2)
+  - windows (v1.44.1)
+  - chef_handler (v1.4.0)
+
+If the directory you provided in the initial prompt contains only `cron` and `chef-client`, then `chef capture` prompts you to add the locations for sources for the remaining three:
+
+```
+Please provide the base checkout path for the following
+cookbooks, or leave blank if no more cookbooks are checked out:
+
+  - logrotate (v1.9.2)
+  - windows (v1.44.1)
+  - chef_handler (v1.4.0)
+
+Checkout Location [none]:
+```
+
+`chef capture` scans the path that you provide and locates the cookbooks that it needs. The command finishes once it accounts for all cookbook sources; it prompts you for another path if it needs more cookbook sources.
+
+### Download Cookbooks from Chef Server
+
+If you do not have access to the original version-controlled source of a cookbook, press return at the prompt for `chef capture`  to use a copy of the cookbook downloaded from the Chef Server.
+
+Upgrading cookbooks from the Chef Server is not an ideal practice. You will make changes to your cookbooks in the course of the upgrade.  Making changes to your cookbooks without the ability to track your changes in version control almost inevitably leads to conflicts between cookbook sources. Reconciling cookbooks with untracked changes is a difficult and time-consuming process. If you find yourself using many cookbooks--or complex cookbooks--downloaded from the Chef Server, it will be worth the effort in the long run to try to track down their version-controlled sources.
```

Comment:
```suggestion
Upgrading cookbooks from the Chef Infra Server is not an ideal practice. You will make changes to your cookbooks in the course of the upgrade.  Making changes to your cookbooks without the ability to track your changes in version control almost inevitably leads to conflicts between cookbook sources. Reconciling cookbooks with untracked changes is a difficult and time-consuming process. If you find yourself using many cookbooks--or complex cookbooks--downloaded from the Chef Server, it will be worth the effort in the long run to try to track down their version-controlled sources.
```